impl/maven-impl/pom.xml

```diff
@@ -112,6 +112,22 @@ under the License.
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-impl</artifactId>
     </dependency>
+    <dependency>
```

Comment:
am unsure are these deps really needed here, maybe only for test?

As if you make them required, you make apache transport or basic connector "mandatory". I'd rather put back these into test scope of maven-impl and let "mima-like use case" user pick deps he needs instead?

Reply:
I've moved out the transports and kept named locks and connectors, as they are really the only implementation.  If needed, we can revisit later.